embulk-core/src/main/java/org/embulk/spi/util/AbstractReporter.java

```diff
@@ -0,0 +1,19 @@
+package org.embulk.spi.util;
+
+import com.google.common.collect.ImmutableMap;
+import org.embulk.spi.ReporterCloseable;
+
+import java.util.Map;
+
+public abstract class AbstractReporter
+        implements ReporterCloseable
+{
+    public final void reportLine(Reporters.ReportLevel level, String line)
```

Comment:
Just a comment: I'm yet very confident we should make these methods `final` actually. For example, some reporter plugins may want to encode binaries in their own encodings.

Starting from `final` would be fine (as I've said), but we may want to leave this intention of ours as a comment.